### PR TITLE
Feature: DEG integration

### PR DESCRIPTION
**Pull request by Leshe4ka**

### CODE REVIEW

File: odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/dto/DataEntityDetailsDto.java [hunk 1]
```diff
@@ -30,73 +31,58 @@
...
-    @Data
-    @Builder
-    @NoArgsConstructor
-    @AllArgsConstructor
-    public static class DataSetDetailsDto {
-        private Long rowsCount;
-        private Long fieldsCount;
-        private Long consumersCount;
-        private Collection<DatasetVersionPojo> datasetVersions;
+    public record DataSetDetailsDto(Long rowsCount, Long fieldsCount, Long consumersCount,
```
Comment (DementevNikita) [inline odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/dto/DataEntityDetailsDto.java:66]: Have you checked if records are properly work with the Jackson?

File: odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/DataEntityRepositoryImpl.java [hunk 2]
```diff
@@ -1272,18 +1110,29 @@ private DataEntityDetailsDto mapDetailsRecord(final Record r) {
...
             .specificAttributes(extractSpecificAttributes(dataEntity, types))
             .namespace(jooqRecordHelper.extractRelation(r, NAMESPACE, NamespacePojo.class))
             .ownership(extractOwnershipRelation(r))
             .types(types)
             .tags(jooqRecordHelper.extractAggRelation(r, AGG_TAGS_FIELD, TagPojo.class))
-            .dataSetDetailsDto(DataSetDetailsDto.builder()
-                .datasetVersions(datasetVersions)
-                .build())
+            .dataSetDetailsDto(new DataEntityDetailsDto.DataSetDetailsDto(
+                null, null, null, datasetVersions))
```
Comment (DementevNikita) [inline odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/DataEntityRepositoryImpl.java:1120]: let's make custom constructor

File: odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/DataEntityRepositoryImpl.java [hunk 3]
```diff
@@ -1366,118 +1215,57 @@ private DataEntityDetailsDto mapDetailsRecord(final Record r) {
...
     @Builder
     @Data
     private static class DataEntitySelectConfig {
         private List<Condition> cteSelectConditions;
         private LimitOffset cteLimitOffset;
-        private List<Condition> joinSelectConditions;
-        private boolean includeDetails;
+        private List<Condition> selectConditions;
+        private boolean includeDimensions;
+        private boolean includeDimensionsAndDetails; // includes both dimensions and details fields
```
Comment (DementevNikita) [inline odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/DataEntityRepositoryImpl.java:1260]: remove the comment

File: odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/service/IngestionServiceImpl.java [hunk 4]
```diff
@@ -200,6 +211,7 @@ private IngestionDataStructure buildStructure(final DataEntityList dataEntityLis
             .map(this::mapTaskRun)
             .collect(Collectors.toList());
 
+        //todo check this
```
Comment (DementevNikita) [inline odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/service/IngestionServiceImpl.java:214]: remove the comment

**Review by soffest — APPROVED** [review]
FE part lgtm.

**Pull request merged**